### PR TITLE
AP_HAL_ChibiOS: remove bl embed in MatekF405-Wing

### PR DESCRIPTION
**Pull request by bugobliterator**
```
Target         Text    Data  BSS     Total  
--------------------------------------------
bin/arduplane  955440  1900  129256  1086596

Target         Text    Data  BSS     Total  
--------------------------------------------
bin/arduplane  943896  1900  129256  1075052
```
saves 11.5KB

### CODE REVIEW

**Comment by nkruzan** [comment]
Does this mean the "arduXXX_with_bl.hex" will not be generated? Used as described here https://ardupilot.org/plane/docs/common-loading-firmware-onto-chibios-only-boards.html#download-the-ardupilot-firmware

**Comment by bugobliterator** [comment]
> Does this mean the "arduXXX_with_bl.hex" will not be generated? Used as described here https://ardupilot.org/plane/docs/common-loading-firmware-onto-chibios-only-boards.html#download-the-ardupilot-firmware

No, this will not do that. This just removes bl copy that we maintain inside the firmware.

**Review by tridge — CHANGES_REQUESTED** [review]
i think this should be an option for the custom fw server, but not enabled yet

File: libraries/AP_HAL_ChibiOS/hwdef/MatekF405-Wing/hwdef.dat [hunk 1]
```diff
@@ -202,3 +202,5 @@ define HAL_BARO_WIND_COMP_ENABLED 0
 define HAL_RUNCAM_ENABLED 0
 define HAL_SPEKTRUM_TELEM_ENABLED 0
 define HAL_SOARING_ENABLED 0
+
+BOOTLOADER_EMBED 0
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/MatekF405-Wing/hwdef.dat:206]:
```suggestion
```

**Review by peterbarker — CHANGES_REQUESTED** [review]
Resolution at the call was add the infrastructure but remove the use of it, I think?

**Comment by peterbarker** [comment]
Needs a rebase somehow

**Comment by peterbarker** [comment]
Could we also fix this to have a single ommit, `AP_HAL_ChibiOS: allow embedded bootloader to be excluded` ?

**Review by tridge — APPROVED** [review]

**Pull request merged**